### PR TITLE
Fixes #9716 and #13004: don't drop the qualifier of quotations at printing time

### LINKED ISSUE #9716: Print Ltac prints invalid Ltac by neglecting to insert constr:
#### Description of the problem

```coq
Require Import Coq.Lists.List.
Ltac foo x := match x with @List.map ?A ?B => constr:((A, B)) end.
Print Ltac foo.
(* Ltac foo x := match x with
              | map (B:=?B) => (A, B)
              end *)
```
This printed Ltac is not valid, as it is missing the `constr:`.

#### Coq Version

8.10+alpha (a83eac8463787c13a2dbd3903baf2b59ca1a4635)

Possibly a duplicate of #6613, though here it actually generates invalid Ltac rather than merely surprising Ltac.

Related: #5169

### LINKED ISSUE #13004: Ltac quotations are not printed.
#### Description
It appears that quotations that are registered through `Tacentries.create_ltac_quotation` are not properly printed. In some cases not printing the quotation results in a valid tactic term, but there are also cases where the resulting string is not a valid tactic. See the two examples below (I'm sure there are more).
```
Require Import Ltac2.Ltac2 Ltac2.Message.

Ltac t := ltac2:(print (of_string "hi")).
Ltac u := ident:(H).

Print t.
Print u.
```

#### Coq Version
```
You are running The Coq Proof Assistant, version 8.13+alpha (September 2020)
```

### LINKED ISSUE #13304: Coqdoc: verbatim mode ends at >> anywhere in the text, not just at begin of line
The reference manual says: "Verbatim material is introduced by a leading << and closed by >> __at the beginning of a line__." See [verbatim](https://coq.inria.fr/refman/using/tools/coqdoc.html?highlight=coqdoc#verbatim).

Actually verbatim mode is ended at a `>>` in the middle of the text, not just at the beginning of a line, which is a sever issue for documenting C code where `>>` is a common operator.

This is with Coq 8.12.0 release on Mac.

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** bug fix

Synchronous overlay for Equations at mattam82/Coq-Equations#322

We applied an easy fix: adding the information of the quotation in the `TacGeneric` node.

Fixes #9716
Fixes #13004

- [X] Added / updated test-suite
- [X] Documentation of the change of API
- [x] Entry added in the changelog

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]
It's a bit hackish but I don't know how to do better.

**Comment by ppedrot** [comment]
@herbelin Is the coq-tools failure related?

**Comment by herbelin** [comment]
> @herbelin Is the coq-tools failure related?

I would bet for no (similar failures as on other PRs), but I cannot assert for sure.

**Comment by ppedrot** [comment]
@herbelin can you rebase just to be sure? If the CI is green I'll merge right after.

**Comment by herbelin** [comment]
New windows failures seem unrelated.

**Comment by ppedrot** [comment]
Crossing my fingers it's the case, let's merge.

**Pull request merged**
